stdlib/public/Cxx/CxxRandomAccessCollection.swift

```diff
@@ -31,6 +31,9 @@ extension UnsafeMutablePointer: UnsafeCxxRandomAccessIterator {}
 public protocol CxxRandomAccessCollection: CxxSequence, RandomAccessCollection {
   override associatedtype RawIterator: UnsafeCxxRandomAccessIterator
   override associatedtype Element = RawIterator.Pointee
+  override associatedtype Index = Int
+  override associatedtype Indices = DefaultIndices<Self>
+  override associatedtype SubSequence = Slice<Self>
```

Comment:
We can also create a special kind of `Slice` for C++ collections which would box the collection to avoid extra copies. I'll do that a bit later.